File: go/robot_bounded_in_circle.go
package leetcode

func isRobotBounded(instructions string) bool {
	x, y, dir := 0, 0, 0
	directions := [][]int{{0, 1}, {1, 0}, {0, -1}, {-1, 0}}

	for _, instruction := range []rune(instructions) {
		if instruction == 'R' {
			dir = (dir + 1) % 4
		} else if instruction == 'L' {
			dir = (dir + 3) % 4
		} else {
			x, y = x+directions[dir][0], y+directions[dir][1]
		}
	}

	return (x == 0 && y == 0) || dir != 0
}

File: python/robot-bounded-in-circle.py
class Solution:
    def isRobotBounded(self, instructions: str) -> bool:
        dirs = [(0, 1), (1, 0), (0, -1), (-1, 0)]
        curr_dir, curr_pos = 0, (0, 0)
        
        for instruction in instructions:
            if instruction == "R":
                curr_dir = (curr_dir + 1) % 4
            elif instruction == "L":
                curr_dir = (curr_dir + 3) % 4
            else:
                curr_pos = (curr_pos[0]+dirs[curr_dir][0], curr_pos[1]+dirs[curr_dir][1])
        return curr_pos == (0, 0) or curr_dir != 0